docs/howto/system_testing.md

```diff
@@ -599,6 +599,80 @@ A sample workflow would look like:
 - Run `elastic-package stack up -d -v`
 - Navigate to the package folder in integrations and run `elastic-package test system -v`
 
+### Running system tests without cleanup (technical preview)
+
+**Disclaimer**: just tested with service deployer `docker`.
+
+By default, `elastic-package test system` command always performs these steps to run tests for a given package:
+1. Setup:
+    - Start the service to be tested with a given configuration (`data_stream/<name>/_dev/test/sytem/test-name-config.yml`) and variant (`_dev/deploy/variants.yml`).
+    - Build and install the package
+    - Create the required resources in Elasticsearch to configure the ingestion of data through the Elastic Agent.
+        - test policy
+        - add package data stream to the test policy
+    - Ensure agent is enrolled.
+    - Ensure that the data stream has no old documents.
+    - Assign policy to the agent.
+    - Ensure there are hits received in Elasticsearch in the package data stream.
+2. Run the tests (validate fields, check transforms, etc.)
+    - Validate fields (e.g. mappings)
+    - Assert number of hit counts.
+    - Check transforms.
+3. Tear Down:
+    - Rollback all the changes in Elasticsearch:
+        - Restore previous policy to the agent.
+        - Delete the test policy.
+        - Uninstall the package.
+    - Stop the service (e.g. container) to be tested.
+    - Wipe all the documents of the data stream.
+
+This process is repeated for each combination of:
+- data stream `D`, if the package is of `integration` type.
+- configuration file defined under `_dev/test/sytem` folder.
+- variant (`_dev/deploy/variants.yml`).
+
+It's possible also to run these steps independently. For that it is required to set which configuration file (`--config-file`)
+and which variant (`--variant`), if any, is going to be used to start and configure the service for these tests.
+
+Then, each step can be run using one of these flags:
+- Run the setup (`--setup`), after this command is executed:
+    - Service container will be kept running.
+    - Elastic Agent is going to be sending documents to Elasticsearch.
+- Run just the tests (`--no-provision`), after this command is executed:
+    - Service container will not be stopped.
+    - Documents in the Data stream will be deleted, so the tests will use the latest documents sent.
```

Comment:
It'd be great to find a way to also reset the agent data, but no need to solve this here, this is not a new issue (https://github.com/elastic/elastic-package/issues/405).